src/OrleansRuntime/Messaging/Gateway.cs

```diff
@@ -538,10 +538,47 @@ private void OnMessageSerializationFailure(Message msg, Exception exc)
                 // we only get here if we failed to serialize the msg (or any other catastrophic failure).
                 // Request msg fails to serialize on the sending silo, so we just enqueue a rejection msg.
                 // Response msg fails to serialize on the responding silo, so we try to send an error response back.
-                Log.Warn(ErrorCode.Messaging_Gateway_SerializationError, String.Format("Unexpected error serializing message {0} on the gateway", msg.ToString()), exc);
+                if (this.Log.IsWarning)
```

Comment:
nit: I don't think we need to check everywhere if logger has **Warning** turned on. Most people set logging equal or above info. this check turns true in most cases.

Reply:
I think we usually check it for Debug or Trace level logs